protocols/floodsub/src/layer.rs

```diff
@@ -354,13 +355,21 @@ impl NetworkBehaviour for Floodsub {
     fn on_connection_handler_event(
         &mut self,
         propagation_source: PeerId,
-        _connection_id: ConnectionId,
+        connection_id: ConnectionId,
         event: THandlerOutEvent<Self>,
     ) {
         // We ignore successful sends or timeouts.
         let event = match event {
-            InnerMessage::Rx(event) => event,
-            InnerMessage::Sent => return,
+            Ok(InnerMessage::Rx(event)) => event,
+            Ok(InnerMessage::Sent) => return,
+            Err(e) => {
+                log::debug!("Failed to send floodsub message: {e}");
+                self.events.push_back(ToSwarm::CloseConnection {
+                    peer_id: propagation_source,
+                    connection: CloseConnection::One(connection_id),
+                });
```

Comment:
Whilst we could do a fancier error handling here now, I don't think improving `libp2p-floodsub` has a high priority. Logging the error and closing the connection is also what we did before so this should be a strict refactoring.